docs/modules/ROOT/pages/explanations/decisions/keycloak.adoc

```diff
@@ -0,0 +1,85 @@
+
+The following section contains the solutions that have been looked at.
+
+[cols="1,1,1,1,1"]
+|===
+|Requirements |https://www.keycloak.org/guides#operator[Official Operator] |https://glasskube.eu/docs/crd-reference/keycloak/[Glasskube Operator] |https://github.com/codecentric/helm-charts/tree/master/charts/keycloak[Codecentric Helmchart] |https://github.com/bitnami/charts/tree/main/bitnami/keycloak[Bitnami Helmchart]
+
+|Standalone and cluster| ✅ | ❌ only standalone | ✅ | ✅
+
+|Customize Keycloak Settings | ✅ | ❌ | ✅ | ✅
+
+|Metrics | ❌ | ❌ | ✅ | ✅
+
+|Use custom image | ✅ | ❌ | ✅ | ✅
+
+|Regular Maintenance | ❌ | ❌ | ❌ | ❌
+
+|Version Upgrades |✅ | ❌ | ✅ | ✅
+
+|===
+
+
+Some additional notes:
+
+The Glasskube operator supports various tools and the keycloak support is currently very minimalistic (see offical docs).
+
+The official keycloak operator officially only supports a handful of settings. However, it has been designed in a way to completely customize keycloak.
+It uses deployments for deploying keycloak.
+
+The codecentric helm chart is very flexible and supports a lot of settings. It uses a statefulset for deploying keycloak.
+
+The bitnami helm chart has the most settings and flexibility. However, the chart works best in conjunction with the bitnami keycloak image and certain settings might not work correctly if using a different image.
+
+== Decision
+
+https://github.com/codecentric/helm-charts/tree/master/charts/keycloak[Codecentric Helmchart].
+
+The instantiation of the Helm Charts will be handled by `+provider-kubernetes+`.
+
+
+Advantages::
+
+* Less complexity than operators, but provides the same features
+* We have been using this helm chart to deploy keycloak for years. We know it works.
+* The helm chart is very flexible and supports a lot of settings
+* The helm chart is actively maintained and updated regularly
+* No need to create a provider for a keycloak operator.
+* Migrating to another helm chart or forking the helm chart if neeeded is easier than replacing an operator.
+
+
+Disadvantages::
+
+* Lack of regular, automated maintenance
+
+
+=== Rational
+
+Both operators are not very mature and lack basic features.
+
+Although it is possible to configure every aspect of the deployment with the official operator using the `+unsupported+` field. It is officially not supported by Keycloak.
+
+The Glasskube operator is very rudimentary and does not support any of the features we need.
+
+The official operator is written in Java and the Glasskube operator is written in Kotlin.
+We lack the expertise to contribute to either of these projects.
+
+As we already have experience deploying services with helm charts and we have been using the codecentric helm chart for years, we are confident that this is the best solution for us.
+
+Furthermore we don't need to write a custom provider as we can use the `+provider-kubernetes+` provider to instantiate the helm chart.
```

Comment:
```suggestion
Furthermore we don't need to write a custom provider as we can use the `+provider-kubernetes+` provider to instantiate the Helm Chart.
```

Why not use `provider-helm`? Or is this a typo?

Reply:
It's a typo. Fixed